src/search.js

```diff
@@ -0,0 +1,56 @@
+import { ToastMark } from '@toast-ui/toastmark';
+
+function getChildNodes(parent) {
+  const nodes = [];
+  let curr = parent.firstChild;
+  while (curr) {
+    nodes.push(curr);
+    curr = curr.next;
+  }
+  return nodes;
+}
+
+/**
+ * @param md a markdown string
+ * @return an array of objects with attributes `name`, `level` corresponding to document headings.
+ * provided in order of occurrence in the document
+ */
+export function searchHeadings(md) {
+  const ast = new ToastMark(md);
```

Comment:
This is beautiful -- and will directly enable tui.citation. @holub008 you the man.